kore/src/Kore/Step/Rule.hs

```diff
@@ -821,6 +825,13 @@ axiomPatternToPattern
   =
     TermLike.mkImplies left right
 
+errorEnsures :: InternalVariable variable => Predicate variable -> a
```

Comment:
How about a TODO to use a different representation for the RulePattern so that this error is not needed anymore?